modules/core/04-channel/keeper/upgrade.go

```diff
@@ -924,54 +916,47 @@ func (k Keeper) abortUpgrade(ctx sdk.Context, portID, channelID string, err erro
 		return errorsmod.Wrap(types.ErrInvalidUpgradeError, "cannot abort upgrade handshake with nil error")
 	}
 
-	upgrade, found := k.GetUpgrade(ctx, portID, channelID)
-	if !found {
-		return errorsmod.Wrapf(types.ErrUpgradeNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
-	}
-
 	channel, found := k.GetChannel(ctx, portID, channelID)
 	if !found {
 		return errorsmod.Wrapf(types.ErrChannelNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
 	}
 
-	// the channel upgrade sequence has already been updated in ChannelUpgradeTry, so we can pass
-	// its updated value.
-	k.restoreChannel(ctx, portID, channelID, channel.UpgradeSequence, channel)
-
 	// in the case of application callbacks, the error may not be an upgrade error.
 	// in this case we need to construct one in order to write the error receipt.
 	upgradeError, ok := err.(*types.UpgradeError)
 	if !ok {
 		upgradeError = types.NewUpgradeError(channel.UpgradeSequence, err)
 	}
 
-	if err := k.WriteErrorReceipt(ctx, portID, channelID, upgrade.Fields, upgradeError); err != nil {
-		return err
-	}
-
+	// the channel upgrade sequence has already been updated in ChannelUpgradeTry, so we can pass
+	// its updated value.
+	k.restoreChannel(ctx, portID, channelID, channel.UpgradeSequence, channel, upgradeError)
 	return nil
 }
 
 // restoreChannel will restore the channel state and flush status to their pre-upgrade state so that upgrade is aborted.
-func (k Keeper) restoreChannel(ctx sdk.Context, portID, channelID string, upgradeSequence uint64, channel types.Channel) types.Channel {
+func (k Keeper) restoreChannel(ctx sdk.Context, portID, channelID string, upgradeSequence uint64, channel types.Channel, err *types.UpgradeError) types.Channel {
```

Comment:
we can probably clean up the whole `restoreChannel` / `abortUpgrade` stuff a bit more in follow ups.

I'm not actually sure if `MustAbortUpgrade` can panic anymore

Reply:
restore can't panic anymore, but abortUpgrade can, but I agree we can clean up the abort/restore logic in a follow up.

Reply:
We could make an issue to track this also if you want, up to you!